File: 프로그래머스/1/12977. 소수 만들기/소수 만들기.js
function solution(nums) {
    var answer = 0;
    const leng = nums.length;
    var tempArr = [];
    
    for (let i = 0; i < leng; i++) {
        for (let j = i+1; j < leng; j++) {
            for (let k = j+1; k < leng; k++) {
                tempArr.push(nums[i] + nums[j] + nums[k]);
            }
        }
    }
    
    // 경우의 수 배열을 탐색하면서
    for (let i = 0; i < tempArr.length; i++) {
        let tempNum = 0; // 각 탐색마다 임시변수를 선언 후
        // 소수를 판별하기 위해 2 ~ i만큼 탐색하면서
        for (let j = 2; j <= tempArr[i]; j++) {
            // 약수를 판별하여 갯수를 더해준다
            if (tempArr[i]%j === 0) {
                tempNum++;
            }
        }
        // 만약 약수의 갯수가 1개이면 정답에 더해준다
        if (tempNum === 1) {
            answer++;
        }
    }
    
    return answer;
}

File: 프로그래머스/1/12977. 소수 만들기/소수 만들기.py
def solution(nums):
    answer = 0
    leng = len(nums) # nums의 길이를 변수로 선언
    tempArr = [] # 경우의 수를 담아줄 임시배열 선언

    # nums의 배열 내에서 3개의 숫자를 뽑은 3중 for문 작성
    for i in range(leng):
        for j in range(i+1, leng):
            for k in range(j+1, leng):
                # 임시배열에 각 경우의 수를 담아준다
                tempArr.append(nums[i] + nums[j] + nums[k])
    
    # 경우의 수 배열을 탐색하면서
    for i in tempArr:
        tempNum = 0 # 각 탐색마다 임시변수를 선언 후
        # 소수를 판별하기 위해 2 ~ i만큼 탐색하면서
        for j in range(2, i+1):
            # 약수를 판별하여 갯수를 더해준다
            if i%j == 0:
                tempNum += 1
        # 만약 약수의 갯수가 1개이면 정답에 더해준다
        if tempNum == 1:
            answer += 1

    return answer